content/v2.0/reference/storage-engine.md

```diff
@@ -0,0 +1,174 @@
+Data is written to InfluxDB using [Line protocol](/) sent via HTTP POST request to the `/write` endpoint.
+Batches of points are sent to InfluxDB.
+Those batches are compressed and written to a WAL for immediate durability.
+The points are also written to an in-memory cache so that newly written points are immediately queryable.
+The cache is periodically written to disk as TSM files.
+As TSM files accumulate, they are combined and compacted into higher level TSM files.
+
+Points can be sent individually; however, for efficiency, most applications send points in batches.
+A typical batch ranges in size from hundreds to thousands of points.
+Points in a POST body can be from an arbitrary number of series.
+Points in a batch do not have to be from the same measurement or tagset.
+
+## Write Ahead Log (WAL)
+
+To ensure durability, we use a Write Ahead Log (WAL).
+<!-- The WAL is a write-optimized storage format that allows for writes to be durable, but not easily queryable -->
+<!-- On the write side, -->
+WAL is a data structure and algorithm that is super simple and powerful.
+It ensures that written data does not disappear when storage engine restarts.
+When a client sends a write request, the following occurs:
+
+1. Write request is appended to the end of the WAL file.
+2. fsync() the data to the file.
+3. Update the in-memory database.
+   <!-- 3. Update the in-memory CACHE? -->
+4. Return success to caller.
+
+fsync() is a system call, so it has a kernel context switch which costs something.
+fsync() takes the file and pushes pending writes all the way through any buffers and caches to disk.
+fsync() is expensive _in terms of time_ but guarantees your data is safe on disk.
+**Important** to batch your points (send in ~2000 points at a time), to fsync() less frequently.
+
+<!-- On read side: -->
+When the storage engine restarts,
+<!-- (if we've pulled the plug, say) -->
+open WAL file and read it back into the in-memory database.
+Answer requests to the /read endpoint.
+
+<!-- ===== V1 material -->
+<!-- TODO is this still true? -->
+<!-- On the file system, the WAL is made up of sequentially numbered files (`_000001.wal`). -->
+<!-- The file numbers are monotonically increasing and referred to as WAL segments. -->
+<!-- When a segment reaches 10MB in size, it is closed and a new one is opened. Each WAL segment stores multiple compressed blocks of writes and deletes. -->
+<!-- Each entry in the WAL follows a [TLV standard](https://en.wikipedia.org/wiki/Type-length-value) with a single byte representing the type of entry (write or delete), a 4 byte `uint32` for the length of the compressed block, and then the compressed block. -->
+
+{{% note%}}
+Once you receive a response to a write request, your data is on disk!
+{{% /note %}}
+
+## Cache
+
+Queries to the storage engine will merge data from the Cache with data from the TSM files.
```

Comment:
maybe start w line 92 then 91 + 93?
Queries execute on cache data, and then the storage engine merges data from the cache and TSM so writes that come in...